Source/Android/jni/Cheats/GraphicsModGroup.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <jni.h>
+
+#include <set>
+#include <vector>
+
+#include "VideoCommon/GraphicsModSystem/Config/GraphicsMod.h"
+#include "VideoCommon/GraphicsModSystem/Config/GraphicsModGroup.h"
+#include "jni/AndroidCommon/AndroidCommon.h"
+#include "jni/AndroidCommon/IDCache.h"
+
+static GraphicsModGroupConfig* GetPointer(JNIEnv* env, jobject obj)
+{
+  return reinterpret_cast<GraphicsModGroupConfig*>(
+      env->GetLongField(obj, IDCache::GetGraphicsModGroupPointer()));
+}
+
+jobject GraphicsModToJava(JNIEnv* env, GraphicsModConfig* mod, jobject jGraphicsModGroup)
+{
+  return env->NewObject(IDCache::GetGraphicsModClass(), IDCache::GetGraphicsModConstructor(),
+                        reinterpret_cast<jlong>(mod), jGraphicsModGroup);
+}
+
+extern "C" {
+
+JNIEXPORT void JNICALL
+Java_org_dolphinemu_dolphinemu_features_cheats_model_GraphicsModGroup_finalize(JNIEnv* env,
+                                                                               jobject obj)
+{
+  delete GetPointer(env, obj);
+}
+
+JNIEXPORT jobjectArray JNICALL
+Java_org_dolphinemu_dolphinemu_features_cheats_model_GraphicsModGroup_getMods(JNIEnv* env,
+                                                                              jobject obj)
+{
+  GraphicsModGroupConfig* mod_group = GetPointer(env, obj);
+
+  std::set<std::string> groups;
+
+  for (const auto& mod : mod_group->GetMods())
+  {
+    if (mod.m_groups.empty())
+      continue;
+
+    for (const auto& group : mod.m_groups)
+    {
+      groups.insert(group.m_name);
+    }
+  }
+
+  std::vector<GraphicsModConfig*> mods;
+
+  for (GraphicsModConfig& mod : mod_group->GetMods())
+  {
+    // Group only mods shouldn't be shown
```

Comment:
Part of me wonders if we should add a `IsModVisibleInUI(const std::set<std::string>& groups)` function to be used in Qt and Android.  Unlikely to add more visibility features but if it was needed, it would be in one place.

Reply:
I've removed some redundant `empty` checks instead. The code now feels short enough that it might be fine to just leave it duplicated.

Reply:
That works.  Thanks!